### PR TITLE
Posting signal preventive maintenance work orders from PostgreSQL database to Knack

### PR DESCRIPTION
**Pull request by JoeyL6**

### CODE REVIEW

**Comment by johnclary** [comment]
@JoeyL6 are you able to resolve conflicts before I add my review? or would you like me to resolve the conflicts?

**Comment by JoeyL6** [comment]
Hi, John,
I've resolved the conflict. Feel free to take a look.

File: transportation-data-publishing/data_tracker/signal_pms_fulcrum.py [hunk 1]
```diff
@@ -0,0 +1,424 @@
...
+    for pgrest_record in pgrest_records_list:
+        modified_date = pgrest_record["MODIFIED_DATE"]
+
+        if modified_date >= last_run_date:
+            pm_payloads.append(pgrest_record)
+
+    return pm_payloads
+
+def replace_pm_records(postgre_records, knack_pm_records, signal_records):
+    """Summary
```
Comment (johnclary) [inline transportation-data-publishing/data_tracker/signal_pms_fulcrum.py]: You've done a good job documenting the other functions. Would you write a brief summary/return definitions for this?

File: transportation-data-publishing/data_tracker/signal_pms_fulcrum.py [hunk 2]
```diff
@@ -0,0 +1,424 @@
...
+            api_key=KNACK_CREDENTIALS["data_tracker_test"]["api_key"],
+            app_id=KNACK_CREDENTIALS["data_tracker_test"]["app_id"],
+            method="update",
+        )
+
+    return len(pm_insert_payloads) + len(pm_update_payloads)
+
+
+def datetime_to_unix_timestamp(dateandtime):
+    """Summary
```
Comment (johnclary) [inline transportation-data-publishing/data_tracker/signal_pms_fulcrum.py]: You can remove the docstring for this very short function.

File: transportation-data-publishing/data_tracker/signal_pms_fulcrum.py [hunk 3]
```diff
@@ -0,0 +1,424 @@
...
+        TYPE: Description
+    """
+    return (
+        int(
+            time.mktime(datetime.strptime(dateandtime, "%Y-%m-%dT%H:%M:%S").timetuple())
+        )
+    ) * 1000
+
+
+def prepare_signals_payloads(payloads):
```
Comment (johnclary) [inline transportation-data-publishing/data_tracker/signal_pms_fulcrum.py]: You've done a good job documenting the other functions. Would you write a brief summary/return definitions for this?

File: transportation-data-publishing/data_tracker/signal_pms_fulcrum.py [hunk 4]
```diff
@@ -0,0 +1,424 @@
+"""Summary
```
Comment (johnclary) [inline transportation-data-publishing/data_tracker/signal_pms_fulcrum.py]: Would you add a short summary of what this script does?

**Comment by johnclary** [comment]
@JoeyL6 thanks for fixing the `set_trace()`. Would you review my other comments in this PR?

**Comment by johnclary** [comment]
@JoeyL6 since this remains unmerged, can you go ahead an update this to use [pypgrest](https://github.com/cityofaustin/pypgrest). I'm happy to get you started.

File: transportation-data-publishing/data_tracker/signal_pms_fulcrum.py [hunk 5]
```diff
@@ -0,0 +1,424 @@
...
+    for payload in pm_insert_payloads:
+
+        insert_res = knackpy.record(
+            payload,
+            obj_key=cfg["knack_pms"]["ref_obj"],
+            api_key=KNACK_CREDENTIALS["data_tracker_test"]["api_key"],
+            app_id=KNACK_CREDENTIALS["data_tracker_test"]["app_id"],
+            method="create",
+        )
+    pdb.set_trace()
```
[the remark this answers is not in the document]
Comment (JoeyL6) [reply transportation-data-publishing/data_tracker/signal_pms_fulcrum.py]: it has been deleted.

File: transportation-data-publishing/config/postgrest/config.py [hunk 6]
```diff
@@ -21,3 +21,7 @@
     "date_field": "published_date",
     "status_date_field": "traffic_report_status_date_time",
 }
+
+SIGNAL_PMS_TO_POSTGRE = {
```
[the remark this answers is not in the document]
Comment (JoeyL6) [reply transportation-data-publishing/config/postgrest/config.py]: sure, it has been removed

File: transportation-data-publishing/data_tracker/signal_pms_fulcrum.py (hunk 2 above)
Comment (JoeyL6) [reply transportation-data-publishing/data_tracker/signal_pms_fulcrum.py]: the doc string for this function has been deleted

File: transportation-data-publishing/data_tracker/signal_pms_fulcrum.py (hunk 3 above)
Comment (JoeyL6) [reply transportation-data-publishing/data_tracker/signal_pms_fulcrum.py]: this function has been documented

**Comment by JoeyL6** [comment]
Hi, John, I've implemented pypgrest on this project. Please feel free to take a look when you have time.

**Pull request merged**